modules/kafka/src/main/resources/application.conf

```diff
@@ -0,0 +1,71 @@
+{
+  "input": {
+    "type": "Kafka"
+    "topicName": "collector-payloads"
+    "bootstrapServers": "localhost:9092"
+    "consumerConf": {
+      "enable.auto.commit": "false"
+    }
+  }
+
+  "output": {
+    "good": {
+      "type": "Kafka"
+      "topicName": "enriched"
+      "bootstrapServers": "localhost:9092"
+      "producerConf": {
+        "acks": "all"
+      }
+    }
+
+    "pii": {
+      "type": "Kafka"
+      "topicName": ""
+      "bootstrapServers": "localhost:9092"
```

Comment:
I put a 👍 but not 100% sure. Could you explain this further @benjben ?

Reply:
Oops sorry there was a mistake in my sentence (I had updated my initial one), it should have been
```
This can't be a default value for that
```
The reason is that default values have to work, and here if we specify just a `topicName` but no `bootstrapServers` it won't work (unless with local brokers but that's only for testing)

Reply:
I made `pii.bootstrapServers` equal to `""` but this makes the app fail with `org.apache.kafka.common.config.ConfigException: No resolvable bootstrap urls given in bootstrap.servers` when we provide topicName without bootstrapServers. -- I'll improve config decoder to beautify this

If this is the behaviour we want to have, I'll empty `good.bootstrapServers` and `bad.bootstrapServers` as well in application.conf cc @benjben 

Please let me know if I misunderstand your point

Reply:
Indeed `bootstrapServers` should be removed from `application.conf` for good and bad, as well as `topicName`, as only values that will always work should be in `application.conf`.

For `pii`:
- We want to specify some default values for `type` and `producerConf`
- To specify default values we need `pii` block to exist in `application.conf`
- If `pii` block exists, it will be parsed and needs to be valid in terms of [this class](https://github.com/snowplow/enrich/blob/enrich-kafka/modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/config/io.scala#L214-L219)
- To be valid, all the non-optional class members have to be defined
- So `bootstrapServers` and `topicName` have to be there
- What we can do is to set them to empty string, and then check and if they are both empty we should remove pii output, exactly like [this is done for Kinesis and PubSub](https://github.com/snowplow/enrich/blob/master/modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/config/ConfigFile.scala#L75-L80)
- Then [here](https://github.com/snowplow/enrich/blob/enrich-kafka/modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/config/io.scala#L263-L264) you need to make the parsing fail if only one of `bootstrapServers` and `topicName` is not empty

Reply:
thanks @benjben ! I just pushed a separate commit to address this, I previously implemented pii removal [here](https://github.com/snowplow/enrich/blob/enrich-kafka/modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/config/ConfigFile.scala#L81-L83) and updated output decoder [here](https://github.com/snowplow/enrich/blob/enrich-kafka/modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/config/io.scala#L263-L264) to fail if one of them is empty

Reply:
I needed to update output decoder to fail only if one of both fields is empty, done in a separate commit